commands/project/view/project_view.go

```diff
@@ -0,0 +1,224 @@
+package view
+
+import (
+	"encoding/base64"
+	"fmt"
+	"github.com/MakeNowJust/heredoc"
+	"github.com/profclems/glab/api"
+	"github.com/profclems/glab/commands/cmdutils"
+	"github.com/profclems/glab/pkg/git"
+	"github.com/profclems/glab/pkg/iostreams"
+	"github.com/profclems/glab/pkg/utils"
+	"github.com/spf13/cobra"
+	"github.com/xanzy/go-gitlab"
+	"strings"
+)
+
+type ViewOptions struct {
+	ProjectID    string
+	ApiClient    *gitlab.Client
+	Web          bool
+	Branch       string
+	Browser      string
+	GlamourStyle string
+
+	IO *iostreams.IOStreams
+}
+
+func NewCmdView(f *cmdutils.Factory) *cobra.Command {
+	opts := ViewOptions{
+		IO: f.IO,
+	}
+
+	var projectViewCmd = &cobra.Command{
+		Use:   "view [repository] [flags]",
+		Short: "View a project/repository",
+		Long:  `Display the description and README of a project or open it in the browser.`,
+		Args:  cobra.MaximumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			repo, err := f.BaseRepo()
+			if err != nil {
+				return err
+			}
+
+			if opts.ProjectID == "" {
+				if len(args) == 1 {
+					opts.ProjectID = args[0]
+				} else {
+					opts.ProjectID = repo.FullName()
+				}
+			}
+
+			if opts.Branch == "" {
+				opts.Branch, err = git.CurrentBranch()
```

Comment:
The Factory struct already comes with the current branch in `f.Branch()`. Please use `f.Branch()` instead of `git.CurrentBranch()`